libs/api/books/src/lib/reading-list.service.ts

```diff
@@ -28,4 +28,22 @@ export class ReadingListService {
       return list.filter(x => x.bookId !== id);
     });
   }
+
+  async finishBook(
+    id: string,
+    finishedDate: string
+  ): Promise<ReadingListItem[]> {
+    this.storage.update(list =>
+      list.map(item =>
+        item.bookId === id
+          ? ({
+              ...item,
+              finishedDate: finishedDate,
+              finished: true
```

Comment:
Are both of these needed? Doesn't the `finishDate` existence indicate that `finished===true`?